src/substrate/filters/enricher.ts

```diff
@@ -66,6 +67,25 @@ export async function Enrich(
         }
       }
 
+      /**
+       * Offences Events
+       */
+      case EventKind.Offence: {
+        const [ offenceKind, opaqueTimeSlot, applied ] = event.data as unknown as [ Kind, OpaqueTimeSlot, boolean ];
+        const reportIds = await api.query.offences.concurrentReportsIndex(offenceKind, opaqueTimeSlot);
+        const offenceDetails: Option<OffenceDetails>[] = await api.query.offences.reports
+          .multi(reportIds.map((reportId) => reportId.toString()));
```

Comment:
You shouldn't necessarily need to convert to string here, it should accept the codec type as-is.

Also this is a fairly heavy-weight API operation -- how often do we expect these events to come in? We wont want to do this very regularly.

Reply:
Offences come in every session/era so this is not an issue.